### PR TITLE
Fix warning about ProposalCard memory leak

### PR DESCRIPTION
**Pull request by tibetsprague**
Fixes https://github.com/daostack/alchemy/issues/1374

Since i couldn't easily reproduce the warning this is a bit of guess, but I think it's the right guess :)

### CODE REVIEW

File: src/components/Proposal/ProposalData.tsx [hunk 1]
```diff
@@ -79,10 +80,14 @@ class ProposalData extends React.Component<IProps, IState> {
     // Don't schedule timeout if its too long to wait, because browser will fail and trigger the timeout immediately
     const millisecondsUntilExpires = closingTime(this.props.data[0]).diff(moment());
     if (!this.state.expired && millisecondsUntilExpires < 2147483647) {
-      setTimeout(() => { this.setState({ expired: true });}, millisecondsUntilExpires);
+      this.expireTimeout = setTimeout(() => { this.setState({ expired: true });}, millisecondsUntilExpires);
```
Comment (dkent600) [inline src/components/Proposal/ProposalData.tsx]: I don't see a reason for this.  `setState` is mean to be invoked within `componentDidMount`.

File: src/components/Proposal/ProposalData.tsx [hunk 2]
```diff
@@ -79,10 +80,14 @@ class ProposalData extends React.Component<IProps, IState> {
...
     }
```
Comment (dkent600) [inline src/components/Proposal/ProposalData.tsx]: I don't see a reasons for this. 1) `setState` is meant to be called from `componentDidMount`,  2) the console msg indicates that the component has not yet mounted, which would not be the case in `componentDidMount`.

File: src/components/Proposal/ProposalData.tsx (hunk 1 above)
Comment (dkent600) [reply src/components/Proposal/ProposalData.tsx]: Are you not able to consistently reproduce accoring to the lin [given here](https://github.com/daostack/alchemy/issues/1374#issuecomment-583672201)?

File: src/components/Proposal/ProposalData.tsx (hunk 2 above)
Comment (dkent600) [reply src/components/Proposal/ProposalData.tsx]: The console message suggests this is happening in `WithSubscriptionComponent`, which calls `setState` in a lot of places.

Comment (dkent600) [reply src/components/Proposal/ProposalData.tsx]: The console message suggests this is happening in `WithSubscriptionComponent`, which calls `setState` in a lot of places.

File: src/components/Proposal/ProposalData.tsx (hunk 1 above)
Comment (tibetsprague) [reply src/components/Proposal/ProposalData.tsx]: @dkent600 what do you mean? this is inside componentDidMount

Comment (tibetsprague) [reply src/components/Proposal/ProposalData.tsx]: and no I was not able to reproduce. I saw the error happen one time but I dont know when.

File: src/components/Proposal/ProposalData.tsx (hunk 2 above)
Comment (tibetsprague) [reply src/components/Proposal/ProposalData.tsx]: that error message usually occurs when the component is unmounted, and it usually indicates that something setup in componentDidMount was not removed/unset/cleared in componentWillUnmount. So in this case for example the timeout expires/fires and the component is no longer mounted so then this error fires. Thus this fix.

Comment (dkent600) [reply src/components/Proposal/ProposalData.tsx]: sorry, misread the code, was thinking that you added a call to setTimeout that wasn't there before, but I see that is not the case.

**Pull request merged**